src/PrestaShopBundle/Form/Admin/Catalog/Category/AbstractCategoryType.php

```diff
@@ -67,51 +71,87 @@ abstract class AbstractCategoryType extends TranslatorAwareType
      */
     private $configuration;
 
+    /**
+     * @var Router
+     */
+    private $router;
+
+    /**
+     * @var CommandBusInterface
+     */
+    private $queryBus;
+
     /**
      * @param TranslatorInterface $translator
      * @param array $locales
      * @param array $customerGroupChoices
      * @param FeatureInterface $multistoreFeature
      * @param ConfigurationInterface $configuration
+     * @param Router $router
+     * @param CommandBusInterface $queryBus
      */
     public function __construct(
         TranslatorInterface $translator,
         array $locales,
         array $customerGroupChoices,
         FeatureInterface $multistoreFeature,
-        ConfigurationInterface $configuration
+        ConfigurationInterface $configuration,
+        Router $router,
+        CommandBusInterface $queryBus
```

Comment:
Query bus should be injected recklessly into a form type And in this case especially, it doesn't seem to be used.